### DISCUSSION TITLE
parseWithZod does not return a typed payload

### DISCUSSION THREAD

**Discussion by gyopiazza** [edmundhung/conform#618] [Q&A]
### Describe the bug and the expected behavior

Hello and thanks for this fantastic library!

I'm using version `1.1.3` of `@conform-to/react` and `@conform-to/zod` but when I use `parseWithZod`, the resulting `payload` doesn't get typed properly. The payload content is correct, it's just the typing that I cannot get to work.

You can see an example in the snippet, am I doing something?

Thank you in advance :)

### Conform version

v1.1.3

### Steps to Reproduce the Bug or Issue

```typescript
export const loginFormSchema = z.object({
  email: z.string().email(),
  password: z.string(),
})

const submission = await parseWithZod(formData, {
  schema: loginFormSchema,
})

if (submission.status === 'success') {
  submission.payload // Gets typed as: Record<string, unknown>
}
```

### Screenshots or Videos

<img width="470" alt="image" src="https://github.com/edmundhung/conform/assets/748898/5eaf116d-dbd7-42dc-99be-83f66da85733">

<img width="678" alt="image" src="https://github.com/edmundhung/conform/assets/748898/1a2405f3-a0a6-45d1-a333-92f2a4e0a4e9">

### Additional context

Using `v5.1.6` of `typescript`

**Comment by edmundhung** [comment]
You can access the parse result with `submission.value` after checking `submission.status === 'success'`. `submission.payload` looks very similar but wasn't typed because it is the serialized value structured based on the name pattern and is accessible regardless of status.

**Reply by marcomuser** [reply]
This however only works if `formData` does not have an intent. In the `onValidate` fn, we cannot get the typed submit values even after checking `submission.status === 'success'` because the intent is `validate` and conform unconditionally returns `undefined` for the `value` prop in this case. I was looking into background syncing a offline-first form in the onValidate handler if the form change was validated successfully but it seems to be a limitation right now that the submission.value is not easily accessible from within the onValidate handler.
